cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetServiceModules.java

```diff
@@ -171,4 +173,30 @@ protected void configure() {
       }
     };
   }
+
+  private static final class DatasetMdsProvider implements Provider<DatasetFramework> {
```

Comment:
As discussed offline, we need a provider to combine default modules and dataset meta table util values.